### ISSUE TITLE
Support ES2015 IdentifierStart and IdentifierPart

### ISSUE THREAD

**Issue by ariya** [eslint/espree#182]
Unlike ES5, ES2015 spec mandates the use of proper Unicode properties for IdentifierName. See section 11.6 for the details.

The following code should result in a valid syntax tree and not an error:

```js
espree.parse('var \\u212E', { ecmaFeatures: { unicodeCodePointEscapes: true } })
```

The character https://codepoints.net/U+212E has the property ID_START which qualifies it to start an identifier. In ES 5 however, this is not a valid identifier since it does not belong to any of the specified categories (Lu, Ll, Lt, Lm, Lo, Nl).

Note: I verified this only with Unicode 7.0.0.

**Comment by nzakas** [comment]
Fixed in 3.0.0.
